CMakeLists.txt

```diff
@@ -43,7 +47,13 @@ if (NOT DEFINED CMAKE_CXX_STANDARD)
   set(CMAKE_CXX_STANDARD 11)
 endif()
 
-if (USE_POSTFIX)
+if (DEFINED PUGIXML_BUILD_DEFINES)
+  foreach(DEFINE ${PUGIXML_BUILD_DEFINES})
+    add_definitions("-D" ${DEFINE})
```

Comment:
I believe this should be using `target_compile_definitions` instead

Reply:
The difference between `add_definitions` and `target_compile_definitions` is that `target_compile_definitions` applies the setting to a specific target, while `add_definitions` applies the definitions to all targets in this folder and any subfolder. 

Since CMake 3.12 there is also `add_compile_definitions` which adds preprocessor definitions to the compiler command line. (In contrast to `add_definitions` which might include both compiler definitions and compiler flags).

I can certainly change into using `target_compile_definitions`, however the question is if the custom options should be applied to all specified targets (`pugixml`, `pugixml-static`, `pugixml-shared` and `check`) or just a subset of them?

Reply:
I think it should be sufficient to add definitions as PUBLIC to pugixml- targets. This will naturally propagate to all targets that use pugixml, and will make it possible to define ABI-breaking macros like WCHAR_MODE safely. The reason why I suggested the change is my understanding is that add_definitions is part of "legacy" CMake feature set, and modern uses recommend target_ functions.

Reply:
Fixed in fc7928a4b74df73024f9d8a9698c9e5b46e18a8a